### PR TITLE
Beta end modal

### PR DESCRIPTION
**Pull request by xpaczka**
Ref #875 
Ref #876 

## What has been done
- Added "Beta has ended" modal on top of the map
- Added "Portal is closed" modal on the onboarding page
- Set flags to prevent users from entering the map, when portal is closed
- Automatically clear onboarded wallet when portal is closed

## What is missing
- Link to claim Galxe NFT
- Endpoint for "Sign up" form

## Testing
- [x] Set `IS_BETA_CLOSED` flag to `true`
- [x] "Beta has ended" modal is on top of the map
- [x] Modal can be closed
- [x] Set `IS_PORTAL_CLOSED` flag to `true`
- [x] Onboarded wallet is cleared from `localStorage`
- [x] "Portal is closed" modal is on the onboarding page
- [x] Population count is not visible
- [x] Is it not possible to connect the wallet

Zrzut ekranu 2023-12-12 o 09 35 26
Zrzut ekranu 2023-12-12 o 09 34 53

### CODE REVIEW

File: src/ui/Island/Modals/BetaEndModal/GetUpdatesForm.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import React, { FormEvent, useState } from "react"
+import Button from "shared/components/Interface/Button"
+
+export default function GetUpdatesForm() {
+  const [emailAddress, setEmailAddress] = useState("")
+
+  const formSubmitHandler = (e: FormEvent) => {
+    e.preventDefault()
+
+    // TODO: logic for sending form
```
Comment (jagodarybacka) [inline src/ui/Island/Modals/BetaEndModal/GetUpdatesForm.tsx:10]: do we want to add a task to handle that later?

File: src/ui/Island/Modals/BetaEndModal/ClaimYourNFT.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        <h2 className="modal_actions_header">Claim your NFT</h2>
+        <p className="modal_actions_description">
+          participate in the beta? See if you are
+          <br />
+          eligible to claim the NFT awards.
+        </p>
+        <Button
+          onClick={(e) => {
+            e.preventDefault()
+            window.open("/", "_blank")
```
Comment (jagodarybacka) [inline src/ui/Island/Modals/BetaEndModal/ClaimYourNFT.tsx:20]: Can we add a link to external links object? if we don't know specific url let's just use `https://galxe.com/taho` for now

File: src/ui/DApps/IslandView.tsx [hunk 3]
```diff
@@ -26,11 +27,24 @@ export default function IslandView() {
...
+  const [betaEndModalVisible, setBetaEndModalVisible] = useState(
+    process.env.IS_BETA_CLOSED === "true"
+  )
+
   return (
     <>
       <FullPageLoader
         loaded={hasLoadedRealmData && hasLoadedSeasonInfo && hasBalances}
       />
+      {process.env.IS_BETA_CLOSED && betaEndModalVisible && (
```
Comment (jagodarybacka) [inline src/ui/DApps/IslandView.tsx:39]:
```suggestion
      {process.env.IS_BETA_CLOSED === "true" && betaEndModalVisible && (
```

File: src/ui/Island/Modals/BetaEndModal/GetUpdatesForm.tsx (hunk 1 above)
Comment (xpaczka) [reply src/ui/Island/Modals/BetaEndModal/GetUpdatesForm.tsx:10]: Added here: https://github.com/tahowallet/dapp/issues/878

**Review by jagodarybacka — CHANGES_REQUESTED** [review]
UI issues:

- [x] there should be vertical line between sections Claim your NFT and Get updated - see figma
- [x] "participate in the beta?" sounds weird to me and looks weird as it starts the line with lowercase character - please confirm we want it like that

image

**Comment by jagodarybacka** [comment]
On a side note - once we have the email thing - would be nice to clear the input on success and show some success message

**Comment by xpaczka** [comment]
> On a side note - once we have the email thing - would be nice to clear the input on success and show some success message

Will take care of it when I start working on the logic when I have the endpoint

**Review by jagodarybacka — APPROVED** [review]
🔥

**Pull request merged**